lib/documents.js

```diff
@@ -2904,78 +2922,91 @@ function readDocs(jobState, val, batchdef, readBatchArray, readerId){
     finishReader(jobState);
     return;
   }
-  readDocumentsImpl.call(jobState.docInstance, val, [batchdef])
-      .result((output) => {
-        jobState.docsReadSuccessfully+= readBatchArray.length;
-        output.forEach(item => {
-          if(jobState.readableObjectMode === 'chunked'){
-            jobState.outputStream.write(JSON.stringify(item));
-          }
-         else {
-            jobState.outputStream.write(item);
-         }
-        });
-        if(jobState.lastBatch){
-          finishReader(jobState);
-        }
-        else {
+  const readDocumentsImplRequest = readDocumentsImpl.call(jobState.docInstance, val, [batchdef]);
+  if(jobState.readableObjectMode === 'chunked'){
+    readDocumentsImplRequest.stream('chunked')
+        .on('error', function(err){
+          readAllDocumentsErrorHandle(jobState, batchdef, readBatchArray, readerId, val, err);
+        })
+        .on('data', function(item){
+          jobState.outputStream.write(item);
+        })
+        .on('end', function(){
           onReadAllDocs(jobState, readerId);
-        }
-      })
+        });
+  }
+  else {
+    readDocumentsImplRequest.result((output) => {
+      jobState.docsReadSuccessfully+= readBatchArray.length;
```

Comment:
I see belatedly that, in chunked mode, the Node.js API cannot report the number of documents read successfully or failed.

Maybe that's something to document?